File: src/main/java/random/telegramhomebot/repository/HostTimeLogRepository.java
package random.telegramhomebot.repository;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import random.telegramhomebot.model.Host;
import random.telegramhomebot.model.HostTimeLog;

import java.sql.Timestamp;
import java.util.List;
import java.util.UUID;

public interface HostTimeLogRepository extends JpaRepository<HostTimeLog, UUID> {
    List<HostTimeLog> findByCreatedDateBetween(Timestamp startDate, Timestamp endDay);

    Page<HostTimeLog> findHostTimeLogByHost(Pageable pageable, Host host);
}

File: src/main/java/random/telegramhomebot/repository/HostTimeLogRepository.kt
package random.telegramhomebot.repository

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import random.telegramhomebot.model.Host
import random.telegramhomebot.model.HostTimeLog
import java.sql.Timestamp
import java.util.*

interface HostTimeLogRepository : JpaRepository<HostTimeLog, UUID?> {
    fun findByCreatedDateBetween(startDate: Timestamp, endDay: Timestamp?): List<HostTimeLog>
    fun findHostTimeLogByHost(pageable: Pageable, host: Host?): Page<HostTimeLog>
}